tests/test_isPrimaryFlag.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+from lsst.utils import getPackageDir
+from lsst.log import Log
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask, CharacterizeImageConfig
+from lsst.pipe.tasks.calibrate import CalibrateTask, CalibrateConfig
+
+
+class IsPrimaryTestCase(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        # Load sample input from disk
+        expPath = os.path.join(getPackageDir("pipe_tasks"), "tests", "data", "v695833-e0-c000-a00.sci.fits")
+        self.exposure = afwImage.ExposureF(expPath)
+        # set log level so that warnings do not display
+        Log.getLogger("calibrate").setLevel(Log.ERROR)
+
+    def tearDown(self):
+        del self.exposure
+
+    def testIsPrimaryFlag(self):
+        """Tests detect_isPrimary column gets added when run.
```

Comment:
Looks like it does more than this, please elaborate briefly, and specify this test is just for single frame processing.

Reply:
Updated the docstring to read `Tests detect_isPrimary column gets added when run, and that sources labelled as detect_isPrimary are not sky sources and have no children.`.